src/mpfao/3D/tdympfao3D_core.c

```diff
@@ -83,22 +83,24 @@ PetscErrorCode TDyComputeGMatrixFor3DMesh(TDy tdy) {
     for (isubcell=0; isubcell<cells->num_subcells[icell]; isubcell++) {
 
       PetscInt subcell_id = icell*cells->num_subcells[icell]+isubcell;
-      PetscInt sOffsetFace = subcells->face_offset[subcell_id];
 
-      PetscInt ii,jj;
+      PetscInt *subcell_face_ids, subcell_num_faces;
+      PetscReal *subcell_face_areas;
+      ierr = TDyMeshGetSubcellFaces(mesh, subcell_id, &subcell_face_ids, &subcell_num_faces); CHKERRQ(ierr);
+      ierr = TDyMeshGetSubcellFaceAreas(mesh, subcell_id, &subcell_face_areas, &subcell_num_faces); CHKERRQ(ierr);
 
-      for (ii=0;ii<subcells->num_faces[subcell_id];ii++) {
+      for (PetscInt ii=0;ii<subcells->num_faces[subcell_id];ii++) {
```

Comment:
Not a big deal, but could you replace `subcells->num_faces[subcell_id]` here with `subcell_num_faces`? Or is that something different?

Reply:
If those things are the same, you could actually replace `subcells->num_faces[subcell_id]` in several places below.